src/kex.c

```diff
@@ -98,6 +98,44 @@
             }                                                              \
     }
 
+/*
+ * Generic Diffie-Hellman computation support.
+ *
+ * DH context should be a _libssh2_bn *.
+ */
+
+#ifndef libssh2_dh_key_pair
```

Comment:
Hi Alexander,
In the meantime, I've lost an access to an iSeries, thus os400. I don't know if I'll regain such an access in the future.
This means I can implement what I suggested on Aug 19, but the os400 implementation will be "blind": I can't test it anymore.
Go or NoGo? Your opinion matters.
Thanks for your reply.